op-batcher/batcher/driver.go

```diff
@@ -566,38 +603,88 @@ func (l *BatchSubmitter) throttlingLoop(wg *sync.WaitGroup, pendingBytesUpdated
 				maxBlockSize = l.Config.ThrottleBlockSize
 			}
 		}
-		var (
-			success bool
-			rpcErr  rpc.Error
-		)
-		err = cl.Client().CallContext(
-			ctx, &success, SetMaxDASizeMethod, hexutil.Uint64(maxTxSize), hexutil.Uint64(maxBlockSize),
-		)
-		if errors.Is(ctx.Err(), context.Canceled) {
-			// If the context was cancelled, our work is done and we expect an error here:
-			// So log it quietly and exit.
-			l.Log.Debug("DA throttling context cancelled")
-			return
-		}
-		if errors.As(err, &rpcErr) && eth.ErrorCode(rpcErr.ErrorCode()).IsGenericRPCError() {
-			l.Log.Error("SetMaxDASize rpc unavailable or broken, shutting down. Either enable it or disable throttling.", "err", err)
-			// We'd probably hit this error right after startup, so a short shutdown duration should suffice.
-			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
-			defer cancel()
-			// Call StopBatchSubmitting in another goroutine to avoid deadlock.
-			go func() {
-				// Always returns nil. An error is only returned to expose this function as an RPC.
-				_ = l.StopBatchSubmitting(ctx)
-			}()
-			return
-		} else if err != nil {
-			l.Log.Error("SetMaxDASize rpc failed, retrying.", "err", err)
-			retryTimer.Reset(retryInterval)
-			return
-		}
-		if !success {
-			l.Log.Error("Result of SetMaxDASize was false, retrying.")
-			retryTimer.Reset(retryInterval)
+
+		// If we have configured DA update endpoints, distribute the configuration
+		if len(daClients) > 0 {
+			l.Log.Info("Distributing DA size configuration", "tx_size", maxTxSize, "block_size", maxBlockSize,
+				"endpoints", len(daClients))
+
+			var wg sync.WaitGroup
+			errors := make(chan error, len(daClients))
+			success := make(chan bool, len(daClients))
+
+			// Set DA size on all endpoints
+			for endpoint, client := range daClients {
+				wg.Add(1)
+				go func(endpoint string, client *rpc.Client) {
+					defer wg.Done()
+
+					succeeded, err := l.setMaxDASizeOnEndpoint(ctx, endpoint, client, maxTxSize, maxBlockSize)
+					if err != nil {
+						errors <- err
+					} else if succeeded {
+						success <- true
+					}
+				}(endpoint, client)
+			}
+
+			// Wait for all operations to complete
+			wg.Wait()
+			close(errors)
+			close(success)
+
+			// Count successes vs. required endpoints
+			successCount := len(success)
+			requiredCount := len(daClients)
+
+			if successCount < requiredCount {
+				l.Log.Error("Failed to distribute DA size configuration to all endpoints",
+					"success", successCount, "required", requiredCount)
+				retryTimer.Reset(retryInterval)
+				return
+			}
+
+			l.Log.Info("Successfully distributed DA size configuration to all endpoints",
+				"tx_size", maxTxSize, "block_size", maxBlockSize)
+		} else {
```

Comment:
The function this is part of has now grown significantly. Would be great to extract the multi-throttling part into a separate function.